src/Factories/FunFunFactory.php

```diff
@@ -518,6 +523,32 @@ function( string $name, array $parameters = [] ): string {
 		$container['payment-types-settings'] = function (): PaymentTypesSettings {
 			return new PaymentTypesSettings( $this->config['payment-types'] );
 		};
+
+		$container['choice_factory'] = function (): ChoiceFactory {
+			return new ChoiceFactory( $this->getFeatureToggle() );
+		};
+
+		$container['campaign_config'] = function (): array {
+			// TODO move to CampaignFactory?
```

Comment:
I don't feel like I understand the purpose / scope of this factory well enough to comment on this, maybe elaborate on this, possibly also in person.